app/api/discount_codes.py

```diff
@@ -97,40 +170,169 @@ def before_create_object(self, data, view_kwargs):
     data_layer = {'session': db.session,
                   'model': DiscountCode,
                   'methods': {
-                      'query': query,
-                      'before_create_object': before_create_object}}
+                    'query': query,
+                    'before_create_object': before_create_object}}
 
 
 class DiscountCodeDetail(ResourceDetail):
     """
     Discount Code detail by id
     """
 
+    def before_get_object(self, view_kwargs):
+        """
+        query method for Discount Code detail
+        :param view_kwargs:
+        :return:
+        """
+        if view_kwargs.get('event_identifier'):
+            event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'event_identifier')
+            view_kwargs['event_id'] = event.id
+
+        if view_kwargs.get('event_id') and current_user.is_admin:
+            event = safe_query(self, Event, 'id', view_kwargs['event_id'], 'event_id')
+            if event.discount_code_id:
+                view_kwargs['id'] = event.discount_code_id
+            else:
+                view_kwargs['id'] = None
+
+        if view_kwargs.get('id'):
+            discount = self.session.query(DiscountCode).filter_by(id=view_kwargs.get('id')).one()
+            if not discount:
+                raise NotFoundError()
+
+            if discount.used_for == 'ticket' and current_user.is_coorganizer(discount.event_id):
+                self.schema = DiscountCodeSchemaTicket
+
+            elif discount.used_for == 'event' and current_user.is_admin:
+                self.schema = DiscountCodeSchemaEvent
+            else:
+                raise UnprocessableEntity({'source': ''},
+                                          "Please check used_for and endpoint and verify your permission")
+
+        else:
+            raise UnprocessableEntity({'source': ''},
+                                      "Please verify your permission. You must be admin to view event\
+                                      discount code details")
+
+    def before_update_object(self, obj, data, view_kwargs):
+        """
+        Method to edit object
+        :param data:
+        :param view_kwargs:
+        :return:
+        """
+        discount = self.session.query(DiscountCode).filter_by(id=view_kwargs.get('id')).one()
+        if not discount:
+            raise NotFoundError()
+
+        if 'used_for' in data:
+            used_for = data['used_for']
+        elif not 'used_for' in data:
+            used_for = discount.used_for
+
+        if discount.used_for == 'ticket' and (current_user.is_coorganizer(view_kwargs.get('event_id'))
+           or current_user.is_organizer(view_kwargs.get('event_id') or current_user.is_admin)) and used_for != 'event':
+            self.schema = DiscountCodeSchemaTicket
+
+        elif discount.used_for == 'event' and current_user.is_admin and used_for != 'ticket':
+            self.schema = DiscountCodeSchemaEvent
```

Comment:
@poush nope. There won't be issue there either. Because it only validates the input data. ?includes is mainly used to see what all data will be shown in response. So it's perfectly fine even in that case. I just checked seeing your comment. 😅